src/Microsoft.Data.SqlClient/tests/FunctionalTests/SqlHelperTest.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.Data.SqlClient.Tests
+{
+    public class SqlHelperTest
+    {
+        private void TimeOutATask()
+        {
+            TaskCompletionSource<bool> tcs = new TaskCompletionSource<bool>();
+            AsyncHelper.WaitForCompletion(tcs.Task, 1); //Will time out as task uncompleted
+            tcs.SetException(new TimeoutException()); //Our task now completes with an error
+        }
+
+        [Fact]
+        public void WaitForCompletion_DoesNotCreateUnobservedException()
+        {
+            var unobservedExceptionHappenedEvent = new AutoResetEvent(false);
+            EventHandler<UnobservedTaskExceptionEventArgs> handleUnobservedException = 
+                (o, a) => { unobservedExceptionHappenedEvent.Set(); };
+
+            TaskScheduler.UnobservedTaskException += handleUnobservedException;
```

Comment:
I don't much like messing with global state from a unit test, but I'm not sure of what other way to test this. I had a look at TaskExceptionHolder (which is what is responsible for raising the event when finalized) but it looks anything along those lines would be messing with the internals of the Task system too much.